src/main/java/hudson/plugins/copyartifact/CopyArtifact.java

```diff
@@ -493,11 +498,9 @@ private boolean perform(Run src, Run<?,?> dst, String expandedFilter, String exp
     public static final class DescriptorImpl extends BuildStepDescriptor<Builder> {
 
         public FormValidation doCheckProjectName(
-                @AncestorInPath AbstractProject anc, @QueryParameter String value) {
-            // Require CONFIGURE permission on this project
-            if (!anc.hasPermission(Item.CONFIGURE)) return FormValidation.ok();
```

Comment:
This permission check applies to current job configuration, not the path to job the method is going to validate. This doesn't make much sense, can be arbitrary job on jenkins. 
I've looked at git history to understand the intent but was imported with first commit.